### PR TITLE
Added 50% button, moved button to top, added button grouping

### PR DESCRIPTION
**Pull request by specialist0x**

### CODE REVIEW

**Comment by rej156** [comment]
lgtm 
damn that's a lot of code change, bless

**Comment by memacius32** [comment]
- Fixed Crash on swap page.
- Added new balance button on stake tri page.

File: src/pages/RemoveLiquidity/index.tsx [hunk 1]
```diff
@@ -510,152 +503,54 @@ export default function RemoveLiquidity({
...
+            />
+            <ColumnCenter>
+              <Plus size="16" color={theme.text2} />
+            </ColumnCenter>
+            <CurrencyInputPanel
+              hideBalance={true}
+              value={formattedAmounts[Field.CURRENCY_B]}
+              onUserInput={onCurrencyBInput}
+              onClickBalanceButton={value =>
+                onUserInput(Field.LIQUIDITY_PERCENT, value === BalanceButtonValueEnum.MAX ? '100' : '50')
```
Comment (specialist0x) [inline src/pages/RemoveLiquidity/index.tsx:544]: Quick thought -- would it be easier if this enum was set up link this instead:

```
const enum BalanceButtonValueEnum {
    MAX = new Percent(100);
    HALF = new Percent(50);
}
```

File: src/components/CurrencyInputPanel/useCurrencyInputPanel.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export default function useCurrencyInputPanel() {
 
     const halfMaxAmountInput = divideCurrencyAmountByNumber(maxAmountInput, 2)
     const atHalfAmount =
-      maxAmountInput != null && halfMaxAmountInput != null ? parsedAmount?.equalTo(halfMaxAmountInput.toExact()) : false
+      maxAmountInput != null && halfMaxAmountInput != null ? parsedAmount?.equalTo(halfMaxAmountInput) : false
```
Comment (specialist0x) [inline src/components/CurrencyInputPanel/useCurrencyInputPanel.ts:59]: Would this work:

```
const atHalfAmount = halfMaxAmountInput != null ? parsedAmount?.equalTo(halfMaxAmountInput) : false
```

File: src/components/BalanceButton/index.tsx [hunk 3]
```diff
@@ -9,26 +9,35 @@ export type Props = {
...
     return null
   }
 
   return (
-    <RowFlat style={{ marginLeft: '4px' }}>
+    <RowFlat style={{ marginLeft: '4px'}} {...rest}>
       <StyledBalanceLeftButton
         disabled={disableHalfButton}
         onClick={() => onClickBalanceButton(BalanceButtonValueEnum.HALF)}
+        height={height ?? 16}
```
Comment (specialist0x) [inline src/components/BalanceButton/index.tsx:33]: Wanna just make `height` an optional arg in `StyledBalanceButton`?

**Comment by specialist0x** [comment]
Looks great!  Since this touches LP and Swaps, can you run through:

1) 50% swap tx
2) 50% add liquidity tx
3) 50% remove liquidity tx
4) 50% stake tx
5) 50% unstake tx
6) I think that's all of them, but any others that I might've missed

Just to make sure the inputted balances are correctly reflected on the block chain

File: src/components/BalanceButton/index.tsx (hunk 3 above)
Comment (memacius32) [reply src/components/BalanceButton/index.tsx]: addressed, replaced inline styles and height prop by styled components

File: src/components/CurrencyInputPanel/useCurrencyInputPanel.ts (hunk 2 above)
Comment (memacius32) [reply src/components/CurrencyInputPanel/useCurrencyInputPanel.ts]: yup!

File: src/pages/RemoveLiquidity/index.tsx (hunk 1 above)
Comment (memacius32) [reply src/pages/RemoveLiquidity/index.tsx:544]: ended up not using this

**Comment by memacius32** [comment]
Fixed decimal bug on CurrencyInputPanel.
Addressed changes

Checked all this cases OK

- [x] 50% swap tx
- [x] 50% add liquidity tx
- [x] 50% remove liquidity tx
- [x] 50% stake tx
- [x] 50% unstake tx

**Pull request merged**